Civi/Api4/WorkflowMessageExample.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Api4;
+
+/**
+ * CiviCRM menu route.
+ *
+ * Provides page routes registered in the CiviCRM menu system.
+ *
+ * Note: this is a read-only api as routes are set via xml files and hooks.
+ *
+ * @see https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_alterMenu/
+ * @searchable none
+ * @since 5.41
+ * @package Civi\Api4
+ */
+class WorkflowMessageExample extends \Civi\Api4\Generic\AbstractEntity {
+
+  /**
+   * @param bool $checkPermissions
+   * @return Generic\AbstractGetAction
+   */
+  public static function get($checkPermissions = TRUE) {
+    return (new Action\WorkflowMessageExample\Get(__CLASS__, __FILE__))
+      ->setCheckPermissions($checkPermissions);
+  }
+
+  /**
+   * @param bool $checkPermissions
+   * @return Generic\BasicGetFieldsAction
+   */
+  public static function getFields($checkPermissions = TRUE) {
+    return (new Generic\BasicGetFieldsAction(__CLASS__, __FUNCTION__, function () {
+      return [
+        [
+          'name' => 'name',
+          'title' => 'Example Name',
+          'data_type' => 'String',
+        ],
+        [
+          'name' => 'title',
+          'title' => 'Example Title',
+          'data_type' => 'String',
+        ],
+        [
+          'name' => 'workflow',
+          'title' => 'Workflow Name',
+          'data_type' => 'String',
+        ],
+        [
+          'name' => 'file',
+          'title' => 'File Path',
+          'data_type' => 'String',
+          'description' => 'If the example is loaded from a file, this is the location.',
+        ],
+        [
+          'name' => 'tags',
+          'title' => 'Tags',
+          'data_type' => 'String',
+          'serialize' => \CRM_Core_DAO::SERIALIZE_SEPARATOR_BOOKEND,
```

Comment:
Oh I get it. You serialize the value in the get action and then rely on the API output formatter to unserialize it back into an array.
That seems a bit silly. Why not just declare `data_type => Array` here?

Reply:
Right, I originally just had it return an array, but (IIRC) the explorer was happier if you coerced it to look more like MySQL data (ie with a serialization format). In a MySQL context, `data_type=>Array` seems nonsensical... but if APIv4 actually works with , then that's cool. I can give that a try...